src/Http/Controllers/Admin/Controller.php

```diff
@@ -3,29 +3,39 @@
 namespace A17\Twill\Http\Controllers\Admin;
 
 use A17\Twill\Exceptions\Handler as TwillHandler;
+use Illuminate\Config\Repository as Config;
 use Illuminate\Contracts\Debug\ExceptionHandler;
+use Illuminate\Contracts\Foundation\Application;
 use Illuminate\Foundation\Auth\Access\AuthorizesRequests;
 use Illuminate\Foundation\Bus\DispatchesJobs;
 use Illuminate\Foundation\Validation\ValidatesRequests;
 use Illuminate\Routing\Controller as BaseController;
+use Illuminate\Support\Arr;
 
 class Controller extends BaseController
 {
     use AuthorizesRequests, DispatchesJobs, ValidatesRequests;
 
-    public function __construct()
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @param Application $app
+     * @param Config $config
+     */
+    public function __construct(Application $app, Config $config)
     {
-        if (config('twill.bind_exception_handler', true)) {
-            app()->singleton(ExceptionHandler::class, TwillHandler::class);
+        if ($config->get('twill.bind_exception_handler', true)) {
+            $app->singleton(ExceptionHandler::class, TwillHandler::class);
```

Comment:
I think it shouldn't since this way you are setting a custom ExceptionHandler only for specific controllers and you are not limiting a user to use TwillHandler. By putting in into Service Provider, you are limiting a user to use it for all exceptions or not to use it at all - so it's kind of a bad practice since this is not a framework, but a independently working package.

@ifox @yanhao-li Prove me if I am wrong, guys.

Reply:
`twill.bind_exception_handler` setting was added in v1.2 to not limit the user, so that was an improvement. But the TwillHandler is designed for Twill controllers, not for Console, other controllers, etc. so we shouldn't move the logic to service providers, you are right.